rainier-core/src/main/scala/com/stripe/rainier/sampler/DualAvg.scala

```diff
@@ -50,38 +45,34 @@ private object DualAvg {
       delta = delta,
       logStepSize = Math.log(stepSize),
       logStepSizeBar = 0.0,
-      avgAcceptanceProb = 0.0,
+      avgError = 0.0,
       iteration = 0,
       shrinkageTarget = Math.log(10 * stepSize)
     )
 
-  def findStepSize(lf: LeapFrog,
-                   params: Array[Double],
-                   delta: Double,
-                   nSteps: Int,
-                   iterations: Int)(implicit rng: RNG): Double = {
-    FINE.log("Finding reasonable initial step size")
-    val stepSize0 = findReasonableStepSize(lf, params)
-    FINE.log("Found initial step size of %f", stepSize0)
-
+  def findStepSize(delta: Double, stepSize0: Double, iterations: Int)(
```

Comment:
we can simplify this method quite a bit and make it much more flexible by separating out the function from stepSize => logAcceptanceProb, and letting the caller supply that. Similarly, we take an initial step size rather than hard coding the findReasonableStepSize call.